### PR TITLE
Ignore properties that are really just functions.

### PR DESCRIPTION
**Pull request by saquibmian**
This ignores things like:
```csharp
public static IFoo FooBar { 
    get { 
        return new Foo(); 
    } 
}
```
...because they're not actually static variables.

### CODE REVIEW

File: src/D2L.CodeStyle.Analysis/TypeSymbolExtensions.cs [hunk 1]
```diff
@@ -41,4 +43,11 @@ public static class TypeSymbolExtensions {
             return args;
         }
     }
+
+    public static class SyntaxNodeExtension {
+        public static bool IsPropertyGetterImplemented(this PropertyDeclarationSyntax syntax) {
```
Comment (travisMiehm) [inline src/D2L.CodeStyle.Analysis/TypeSymbolExtensions.cs]: Tests for this?

**Comment by saquibmian** [comment]
Ugh stupid TortoiseGit. Hang on.

File: tests/D2L.CodeStyle.Analysis.UnitTests/D2L.CodeStyle.Analysis.UnitTests.csproj [hunk 2]
```diff
@@ -119,6 +121,9 @@
     <Analyzer Include="..\..\packages\Microsoft.CodeAnalysis.Analyzers.1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.Analyzers.dll" />
     <Analyzer Include="..\..\packages\Microsoft.CodeAnalysis.Analyzers.1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll" />
   </ItemGroup>
+  <ItemGroup>
+    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
```
Comment (saquibmian) [inline tests/D2L.CodeStyle.Analysis.UnitTests/D2L.CodeStyle.Analysis.UnitTests.csproj]: kill

File: src/D2L.CodeStyle.Analysis/TypeSymbolExtensions.cs (hunk 1 above)
Comment (saquibmian) [inline src/D2L.CodeStyle.Analysis/TypeSymbolExtensions.cs]: Added.

**Comment by saquibmian** [comment]
@travisMiehm good?

File: tests/D2L.CodeStyle.Analysis.UnitTests/TestBase.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+
+namespace D2L.CodeStyle.Analysis {
+
+    public class TestBase {
```
Comment (travisMiehm) [inline tests/D2L.CodeStyle.Analysis.UnitTests/TestBase.cs]: I don't like this as a base class at all.  Go with static methods.

Comment (saquibmian) [inline tests/D2L.CodeStyle.Analysis.UnitTests/TestBase.cs]: Why? I prefer this base class over static methods...

**Pull request merged**